paasta_tools/utils.py

```diff
@@ -1700,3 +1701,45 @@ def prompt_pick_one(sequence, choosing):
         sys.exit(1)
     else:
         return result
+
+
+def retry(ExceptionToCheck, tries=4, delay=3, backoff=2, logger=None):
```

Comment:
As discussed, https://github.com/invl/retry / https://pypi.python.org/pypi/retry/ can be reused here.